### PR TITLE
Update tooling to targetSdkVersion 29 and compileSdkVersion 29

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/414730916066338/1187618988784913/f
Tech Design URL: 
CC: 

**Description**:
Bumping the target and compile SDK versions resulted in some compilation errors, mostly around stricter nullability checks. 

**Steps to test this PR**:
There is the possibility of a subtle problem being introduced anywhere though, so need to verify the whole app is still working as expected. 

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/browser/logindetection/NavigationAwareLoginDetector.kt [hunk 1]
```diff
@@ -91,15 +91,20 @@ class NextPageLoginDetection @Inject constructor() : NavigationAwareLoginDetecto
...
         Timber.i("LoginDetectionDelegate ${loginAttemptEvent.url} vs $forwardedToUrl")
         if (loginURI.host != forwardedToUri.host || loginURI.path != forwardedToUri.path) {
-            loginEventLiveData.value = LoginDetected(loginURI.host, forwardedToUri.host)
+            val loginUriHost = loginURI.host
+            val forwardedToUriHost = forwardedToUri.host
+
+            if (loginUriHost != null && forwardedToUriHost != null) {
+                loginEventLiveData.value = LoginDetected(loginUriHost, forwardedToUriHost)
+            }
+
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/browser/logindetection/NavigationAwareLoginDetector.kt:107]: @cmonfortep, as discussed, you probably want to make a few small changes here?

Comment (cmonfortep) [reply app/src/main/java/com/duckduckgo/app/browser/logindetection/NavigationAwareLoginDetector.kt]: Just pushed a change related to this part 🚀

**Comment by CDRussell** [comment]
@cmonfortep, thanks for your changes; they're good with me. I'll throw the whole PR back at you for a final review if you don't mind.

**Comment by malmstein** [comment]
@CDRussell this can be merged and I'll add it to the next release unless something else is stopping it?

**Comment by CDRussell** [comment]
I had originally planned to put this out in its own release but then quite a few releases started happening; i'm happy to include it with other PRs though. Thanks

**Pull request merged**